### ISSUE TITLE
Build does multiple builds instead of just one - sometimes....

### ISSUE THREAD

**Issue by duglin** [knative/build#575]
/area API
/kind bug

## Expected Behavior

When I change the `build` section of Service I expect just one build to be kicked off.

## Actual Behavior

About 1/2 the time I see more than one build (pod) kicked off. Usually I see 2 build pods, but I think I've even seen 3 before but can't prove it :-)

## Steps to Reproduce the Problem

See: https://github.com/duglin/helloworld for the complete flow.
But if you basically use this yaml:
```
apiVersion: serving.knative.dev/v1alpha1
kind: Service
metadata:
  name: helloworld
spec:
  runLatest:
    configuration:
      build:
        apiVersion: build.knative.dev/v1alpha1
        kind: Build
        metadata:
          annotations:
            trigger: "15"
        spec:
          serviceAccountName: build-bot
          source:
            git:
              revision: master
              url: https://github.com/duglin/helloworld
          template:
            name: kaniko
            arguments:
            - name: IMAGE
              value: index.docker.io/duglin/helloworld
      revisionTemplate:
        spec:
          container:
            image: duglin/helloworld
          containerConcurrency: 1
```
with the appropriate SA filled in, and then do:
`kubectl edit ksvc/helloworld` and change the `trigger` value you should see it happen. But it might take a few edits to see it.

**Comment by ImJasonH** [comment]
This sounds like https://github.com/knative/build/issues/518 which was fixed in 0.4 -- if this is still happening we need to figure out why and make it stop.

**Comment by duglin** [comment]
Closing - no longer needed
